raiden/transfer/mediated_transfer/initiator_manager.py

```diff
@@ -196,11 +263,18 @@ def handle_transferrefundcancelroute(
         pseudo_random_generator: random.Random,
         block_number: BlockNumber,
 ) -> TransitionResult:
+    initiator_state = payment_state.initiator_transfers.get(state_change.transfer.lock.secrethash)
+    if not initiator_state:
+        return TransitionResult(payment_state, list())
+
+    channel_identifier = initiator_state.channel_identifier
+    channel_state = channelidentifiers_to_channels.get(channel_identifier)
+
+    if not channel_state:
+        return TransitionResult(payment_state, list())
 
-    channel_identifier = payment_state.initiator.channel_identifier
-    channel_state = channelidentifiers_to_channels[channel_identifier]
     refund_transfer = state_change.transfer
-    original_transfer = payment_state.initiator.transfer
+    original_transfer = initiator_state.transfer
 
     is_valid_lock = (
```

Comment:
I wonder if we really need this checks, `channel.handle_receive_refundtransfercancelroute` should be doing all of this and better.